storage/registers_delta_test.go

```diff
@@ -0,0 +1,221 @@
+		require.NoError(t, err)
+
+		v, err := r.GetValue(owner, key)
+		require.NoError(t, err)
+		require.Equal(t, value2, v)
+	})
+
+	runDB("set-unset-get register", t, func(t *testing.T, db *pebbleStorage.Storage) {
+		t.Parallel()
+
+		s := pebbleStorage.NewRegisters(db, ownerAddress)
+		r := storage.NewRegistersDelta(0, s)
+
+		err := r.SetValue(owner, key, value1)
+		require.NoError(t, err)
+
+		err = r.SetValue(owner, key, nil)
+		require.NoError(t, err)
+
+		v, err := r.GetValue(owner, key)
+		require.NoError(t, err)
+		// not actually nil, but empty
+		require.Len(t, v, 0)
+	})
+
+	runDB("set-next-get register", t, func(t *testing.T, db *pebbleStorage.Storage) {
+		t.Parallel()
+
+		s := pebbleStorage.NewRegisters(db, ownerAddress)
+		r := storage.NewRegistersDelta(0, s)
+
+		err := r.SetValue(owner, key, value1)
+		require.NoError(t, err)
+
+		err = commit(t, db, r, s)
+		require.NoError(t, err)
+
+		r = storage.NewRegistersDelta(1, s)
+
+		v, err := r.GetValue(owner, key)
+		require.NoError(t, err)
+		require.Equal(t, value1, v)
+	})
+
+	runDB("set-next-set-next-get register", t, func(t *testing.T, db *pebbleStorage.Storage) {
+		t.Parallel()
+
+		s := pebbleStorage.NewRegisters(db, ownerAddress)
+		r := storage.NewRegistersDelta(0, s)
+
+		err := r.SetValue(owner, key, value1)
+		require.NoError(t, err)
+
+		err = commit(t, db, r, s)
+		require.NoError(t, err)
+
+		r = storage.NewRegistersDelta(1, s)
+
+		err = r.SetValue(owner, key, value2)
+		require.NoError(t, err)
+
+		err = commit(t, db, r, s)
+		require.NoError(t, err)
+
+		r = storage.NewRegistersDelta(2, s)
+
+		v, err := r.GetValue(owner, key)
+		require.NoError(t, err)
+		require.Equal(t, value2, v)
+	})
+
+	runDB("set-next-unset-next-get register", t, func(t *testing.T, db *pebbleStorage.Storage) {
+		t.Parallel()
+
+		s := pebbleStorage.NewRegisters(db, ownerAddress)
+		r := storage.NewRegistersDelta(0, s)
+
+		err := r.SetValue(owner, key, value1)
+		require.NoError(t, err)
+
+		err = commit(t, db, r, s)
+		require.NoError(t, err)
+
+		r = storage.NewRegistersDelta(0, s)
+
+		err = r.SetValue(owner, key, nil)
+		require.NoError(t, err)
+
+		err = commit(t, db, r, s)
+		require.NoError(t, err)
+
+		r = storage.NewRegistersDelta(1, s)
+
+		v, err := r.GetValue(owner, key)
+		require.NoError(t, err)
+		// not actually nil, but empty
+		require.Len(t, v, 0)
+	})
+
+	runDB("get with wrong owner", t, func(t *testing.T, db *pebbleStorage.Storage) {
+		t.Parallel()
+
+		s := pebbleStorage.NewRegisters(db, ownerAddress)
+		r := storage.NewRegistersDelta(0, s)
+
+		_, err := r.GetValue(owner2, key)
+		require.Error(t, err)
+	})
+
+	runDB("commit with wrong owner", t, func(t *testing.T, db *pebbleStorage.Storage) {
+		t.Parallel()
+
+		s := pebbleStorage.NewRegisters(db, ownerAddress)
+		r := storage.NewRegistersDelta(0, s)
+
+		err := r.SetValue(owner2, key, value1)
+		require.NoError(t, err)
+
+		err = commit(t, db, r, s)
+		require.Error(t, err)
+	})
```

Comment:
Is there a test case to prove the SetValue doesn't save the changes to storage until commit is called? So that we ensure script queries on that height won't see updated values before they are committed.